### ISSUE TITLE
JACPTS() generates nodes and weights containing NaNs

### ISSUE THREAD

**Issue by h2li** [chebfun/chebfun#2134]
I am using Gauss-Jacobi quadrature to calculate the expectation of some function with respect to the Beta distribution. When I call the `jacpts` function to generate the nodes and weights for quadrature, I find the resulting variables contain a lot of `NaN` values.

Because I am dealing with a Beta distribution with alpha = 250 and beta = 170, I try
```
>> [x,w] = jacpts(200,249,169);
>> [sum(isnan(x)) sum(isnan(w))]
```
then I get
```
ans =

   190   200
```
Increasing the number of nodes to 2000 yields a node vector with less NaNs but still a full NaN vector of weights.
```
>> [x,w] = jacpts(2000,249,169);
>> [sum(isnan(x)) sum(isnan(w))]

ans =

           2        2000
```
Is this because of the inaccurate results when alpha and beta are greater than 5? Is there any way to compute the nodes and weights for very large alpha and beta?

For your information, I used MATLAB R2014b & Chebfun 5.6.0 on Windows 7.

**Comment by nickhale** [comment]
Hi @h2li 

I assume you saw the warning
```
>> [x,w] = jacpts(20,249,169);
Warning: ALPHA~=BETA and MAX(ALPHA, BETA) > 5. Results may not be accurate.
```
The reason for this is that the methods used for computing the Gauss-Jacobi points in Chebfun are based on asymptotic results only valid for small-ish values of `alpha` and `beta`.

The `'gw'` option (for Golub-Welsch) *should* be able to cope better in this regime, but it was hitting an overflow error for these large values. #2135 is a pull request for a fix to this.

On that branch, I now get
```
>> [x,w] = jacpts(200,249,169, 'gw'); [sum(isnan(x)) sum(isnan(w))]
Warning: ALPHA~=BETA and MAX(ALPHA, BETA) > 5. Results may not be accurate. 
> In jacpts (line 64) 
ans =
     0     0
```

**Comment by h2li** [comment]
Hi, @nickhale, the fix #2135 and `'gw'` option work great. Thanks!

But I noticed that the weights are normalized when `'gw'` option is used (i.e. `sum(w)' is equal to 1) but the weights from default algorithm are not. Is this for some specific reason?

**Comment by nickhale** [comment]
That was a bug. It was fixed recently in this pull request: #2126
